tests fail with xarray 2022.6.0

The following text raise an IndexError exception

```
FAILED tests/aeroval/test_aeroval_HIGHLEV.py::test_ExperimentOutput__FILES[cfgexp1-chk_files0] - IndexError: dimension coordinat...
FAILED tests/aeroval/test_aeroval_HIGHLEV.py::test_ExperimentOutput__FILES[cfgexp2-chk_files1] - IndexError: dimension coordinat...
FAILED tests/aeroval/test_aeroval_HIGHLEV.py::test_ExperimentOutput__FILES[cfgexp4-chk_files2] - IndexError: dimension coordinat...
FAILED tests/aeroval/test_aeroval_HIGHLEV.py::test_reanalyse_existing[cfgexp4] - IndexError: dimension coordinate 'data_source' ...
FAILED tests/aeroval/test_aeroval_HIGHLEV.py::test_superobs_different_resolutions[cfgexp4] - IndexError: dimension coordinate 'd...
FAILED tests/aeroval/test_experiment_output.py::test_ExperimentOutput_delete_experiment_data_CFG1[cfgexp1] - IndexError: dimensi...
FAILED tests/aeroval/test_experiment_output.py::test_Experiment_Output_clean_json_files_CFG1[cfgexp1] - IndexError: dimension co...
FAILED tests/aeroval/test_experiment_output.py::test_Experiment_Output_clean_json_files_CFG1_INVALIDMOD[cfgexp1] - IndexError: d...
FAILED tests/aeroval/test_experiment_output.py::test_Experiment_Output_clean_json_files_CFG1_INVALIDOBS[cfgexp1] - IndexError: d...
FAILED tests/aeroval/test_experiment_processor.py::test_ExperimentProcessor_run[cfgexp1] - IndexError: dimension coordinate 'dat...
FAILED tests/aeroval/test_experiment_processor.py::test_ExperimentProcessor_run[cfgexp2] - IndexError: dimension coordinate 'dat...
FAILED tests/aeroval/test_experiment_processor.py::test_ExperimentProcessor_run[cfgexp3] - IndexError: dimension coordinate 'dat...
FAILED tests/aeroval/test_experiment_processor.py::test_ExperimentProcessor_run[cfgexp4] - IndexError: dimension coordinate 'dat...
```

for example
```
$ test_ExperimentOutput__FILES[cfgexp1-chk_files0]
[...]
E               IndexError: dimension coordinate 'data_source' conflicts between indexed and indexing objects:
E               ['AeronetSunV3L2Subset.daily' 'TM5-met2010_CTRL-TEST']
E               vs.
E               AeronetSunV3L2Subset.daily

.venv/lib/python3.10/site-packages/xarray/core/coordinates.py:464: IndexError
[...]
```

and
```
$ pytest tests/aeroval/test_experiment_processor.py::test_ExperimentProcessor_run[cfgexp1]
[...]
E               IndexError: dimension coordinate 'data_source' conflicts between indexed and indexing objects:
E               ['AeronetSunV3L2Subset.daily' 'TM5-met2010_CTRL-TEST']
E               vs.
E               AeronetSunV3L2Subset.daily

.venv/lib/python3.10/site-packages/xarray/core/coordinates.py:464: IndexError
[...]
```

--- a/pyaerocom/aeroval/coldatatojson_helpers.py
+++ b/pyaerocom/aeroval/coldatatojson_helpers.py
@@ -1078,7 +1078,8 @@ def _calc_temporal_corr(coldata):
     # Use only sites that contain at least 3 valid data points (otherwise
     # correlation will be 1).
     obs_ok = arr[0].count(dim="time") > 2
-    arr = arr.where(obs_ok, drop=True)
+    arr[0] = arr[0].where(obs_ok, drop=True)
+    arr[1] = arr[1].where(obs_ok, drop=True)
     if np.prod(arr.shape) == 0:
         return np.nan, np.nan
     corr_time = xr.corr(arr[1], arr[0], dim="time")

